apps/modernization-ui/src/apps/page-builder/condition/search/ConditionTable.tsx

```diff
@@ -73,8 +73,9 @@ export const ConditionTable = ({ conditions, isLoading, onSelectionChange, onSor
     const [selected, setSelected] = useState<number[]>([]);
 
     useEffect(() => {
+        setSelected([]);
         setTableRows(conditions.map(asTableRow));
-    }, [conditions]);
+    }, [JSON.stringify(conditions)]);
```

Comment:
just curiious, why the stringify?

Reply:
conditions here is an array of Condition objects
```typescript
type Condition = {
    coinfectionGroup?: string;
    conditionFamily?: string;
    id: string;
    name: string;
    nndInd?: string;
    page?: string;
    programArea: string;
    status: string;
}
```

Using an array as a dependency in a `useEffect` gets wonky. It will trigger a lot more updates than needed. See [Stackoverflow](https://stackoverflow.com/questions/59467758/passing-array-to-useeffect-dependency-list) for where i stole this from